contracts/os/andromeda-kernel/src/ibc.rs

```diff
@@ -145,7 +147,32 @@ pub fn do_ibc_packet_receive(
     );
     match packet_msg {
         IbcExecuteMsg::SendMessage { amp_packet } => {
-            execute_env.amp_ctx = Some(amp_packet.clone());
+            // Try to pull the username's address from the packet
+            match amp_packet.ctx.get_origin_username() {
+                Some(username) => {
+                    // Check if username is registered
+                    let vfs_address = KERNEL_ADDRESSES.load(execute_env.deps.storage, VFS_KEY)?;
+                    let msg = VFSQueryMsg::GetAddressFromUsername { username };
+                    let username_addr =
+                        execute_env
+                            .deps
+                            .querier
+                            .query::<Addr>(&cosmwasm_std::QueryRequest::Wasm(
+                                cosmwasm_std::WasmQuery::Smart {
+                                    contract_addr: vfs_address.to_string(),
+                                    msg: to_json_binary(&msg)?,
+                                },
+                            ));
```

Comment:
This should also be possible with a raw query via `AOSQuerier` returning an `Option<String>`